src/File Input/index.tsx

```diff
@@ -0,0 +1,174 @@
+  const [isSuccess, setIsSuccess] = useState(false)
+  const [selectedFile, setSelectedFile] = useState('')
+
+  return (
+    <FileInputContainer isSuccess={isSuccess}>
+      {isSuccess ? (
+        <Success
+          isLink={isLink}
+          setIsLink={setIsLink}
+          selectedFile={selectedFile}
+          setIsSuccess={setIsSuccess}
+          setSelectedFile={setSelectedFile}
+        />
+      ) : (
+        <UploadArea
+          link={link}
+          text={title}
+          selectedFile={selectedFile}
+          setIsLink={setIsLink}
+          setIsSuccess={setIsSuccess}
+          setSelectedFile={setSelectedFile}
+        />
+      )}
+    </FileInputContainer>
+  )
+}
+
+const Success: React.FC<HelperProps> = ({
+  isLink,
+  setIsSuccess,
+  selectedFile,
+  setSelectedFile,
+}) => {
+  return (
+    <ContentContainer>
+      <DeleteContainer>
+        <DeleteButton
+          onClick={() => {
+            setIsSuccess(false)
+            setSelectedFile('')
+          }}
+        >
+          <DeleteIcon />
+        </DeleteButton>
+      </DeleteContainer>
+
+      <SuccessImageContainer>
+        <SuccessImg />
+      </SuccessImageContainer>
+
+      <SuccessMessageContainer>
+        <SuccessMessage>File telah terupload!</SuccessMessage>
+
+        <FileUploaded>
+          {isLink ? <LinkIcon /> : <FileIcon />}
+          <FileName>{selectedFile}</FileName>
+        </FileUploaded>
+      </SuccessMessageContainer>
+    </ContentContainer>
+  )
+}
+
+const UploadArea: React.FC<HelperProps> = ({
+  link,
+  text,
+  selectedFile,
+  setIsLink,
+  setIsSuccess,
+  setSelectedFile,
+}) => {
+  const onDrop = useCallback((acceptedFiles: any) => {
+    console.log(acceptedFiles)
+    setSelectedFile(acceptedFiles[0].name)
+    setIsSuccess(true)
+  }, [])
+
+  const handleChange = (event: { target: { value: any } }) => {
+    setSelectedFile(event.target.value)
+  }
+
+  const { getRootProps, open, getInputProps } = useDropzone({
+    onDrop,
+    noClick: true,
+    multiple: false,
+    noKeyboard: true,
+  })
+
+  return (
+    <ContentContainer {...getRootProps()}>
+      <Dropzone {...getInputProps()}></Dropzone>
+      <UploadImg />
+      <DropzoneTextContainer>
+        <PrimaryMessageContainer>
+          Drag atau
+          <OpenButton onClick={open}>upload</OpenButton>
+          file kamu di sini
+        </PrimaryMessageContainer>
+        <SecondaryMessage>{text}</SecondaryMessage>
+      </DropzoneTextContainer>
+
+      {link && (
+        <URLContainer>
+          <LineContainer>
+            <Line></Line>
+            atau
+            <Line></Line>
+          </LineContainer>
+
+          <Message>cantumkan link Google Drive</Message>
+
+          <URLInputContainer>
+            <Input
+              type="url"
+              placeholder="Link Google Drive"
+              onChange={handleChange}
+              value={selectedFile}
+            />
+            <SaveButton
```

Comment:
saran gue buat yang `<SaveButton>` mending make yang udah jadi component aja, karena kita bisa modif stylenya lebih gampang.
speaking of that, gue saranin edit jadi semacam ini:
```tsx
 <Button
    disabled={selectedFile == ''}
    onClick={() => {
        setIsSuccess(true)
        setIsLink(true)
    }}
>
    <SaveIcon />
</Button>
```

jadi nanti kemungkinannya bakalan kayak gini:

![image](https://user-images.githubusercontent.com/70869295/213257743-50be8aff-1682-4243-8ba1-d50ab99080b1.png)
![image](https://user-images.githubusercontent.com/70869295/213257790-0c34eda4-921e-40ba-92cc-738071eba486.png)

(oiya sekalian di gue uploadnya kurang margin gtt, kalo di storybook aman tapi pas gue nyoba make yang `yarn local` terus nyoba di suatu nextjs project dia kayak yang di ss)